sql/create.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+// Deletes the existing database in this file
+func deleteDatabase(databasePath string) {
+	fmt.Println("Deleting database", databasePath)
+	os.Remove(databasePath)
+}
+
+// Creates an empty SQLite database in an space-optimized format
+// that allows for massively quick querying of a specific address
+// to get all the (blockIndex, transactionIndex) pairs it's been
+// involved in
+func createEmptyDatabase(databasePath string) {
+	db, err := sql.Open("sqlite3", databasePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+
+	sqlStmt := `
+	CREATE TABLE addresses (
+		addressID 	INTEGER NOT NULL PRIMARY KEY,
+		address    TEXT NOT NULL,
+		UNIQUE(address)
+	);
+	CREATE UNIQUE INDEX idx_addresses_address on addresses(address);
+	CREATE TABLE IF NOT EXISTS txs (
+		txAddressID INTEGER,
+		blockIndex INT, 
+		transactionIndex INT,
+		FOREIGN KEY(txAddressID) REFERENCES addresses(addressID)
+	);
+	CREATE INDEX IF NOT EXISTS idx_txs_address ON txs (txAddressID);
+	`
+	fmt.Println("Creating DB")
+	_, err = db.Exec(sqlStmt)
+	if err != nil {
+		log.Printf("%q: %s\n", err, sqlStmt)
+		return
+	}
+}
+
+type Appearance struct {
+	address          string
+	blockIndex       int
+	transactionIndex int
+}
+
+// Given a TrueBlocks index file in the format to be found at the
+// link below, this file returns all the appearances that are in that
+// specific chunk of the index
+// https://gateway.pinata.cloud/ipfs/Qmart6XP9XjL43p72PGR93QKytbK8jWWcMguhFgxATTya2
+func readFile(path string) ([]Appearance, error) {
+	var appearances []Appearance
+
+	f, err := os.Open(path)
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+
+	headerByteSlice := make([]byte, 44)
+	_, err = f.Read(headerByteSlice)
+	if err != nil {
+		log.Fatal(err)
+	}
+	nAddresses := binary.LittleEndian.Uint32(headerByteSlice[44-8 : 44-4])
+	nAppearances := binary.LittleEndian.Uint32(headerByteSlice[44-4:])
+
+	log.Printf("Number Addresses: %s\n", nAddresses)
+	log.Printf("Number Appearances: %s\n", nAppearances)
+
+	addressByteSlice := make([]byte, 28*nAddresses)
+	_, err = f.Read(addressByteSlice)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	appearanceByteSlice := make([]byte, 8*nAppearances)
+	_, err = f.Read(appearanceByteSlice)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for i := 0; int64(i) < int64(nAddresses); i++ {
+		// TODO: I am not sure this hex encoding is correct, so check it
```

Comment:
We can use the package. That way an optimization there will benefit here, plus it works and is well tested.